velox/functions/prestosql/StringFunctions.h

```diff
@@ -494,4 +494,76 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+template <typename T>
+struct FromBigEndian32 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<int32_t>& result,
+      const null_free_arg_type<Varbinary>& input) {
+    static const auto kTypeLength = sizeof(int32_t);
+    VELOX_USER_CHECK(
+        input.size() == kTypeLength,
+        "Expected 4-byte input, but got instead: {}",
+        input.size());
+    auto buffer = input.data();
+    result = ((int32_t)buffer[0] & 0xff) << 24 |
+        ((int32_t)buffer[1] & 0xff) << 16 | ((int32_t)buffer[2] & 0xff) << 8 |
```

Comment:
Please abstract ((int32_t)buffer[i] & 0xff) as a function and use it in all places in this file.

Reply:
I thought about having a function but I chose to write this explicitly because I was afraid the function call might not be inlined.

Reply:
`((int32_t)buffer[0] & 0xff)` can be a macro something like:
`MASK_BYTE_AT(buffer[0]) << 24`. May be with a better naming.

Reply:
Thanks for the suggestions. Could you explain why this part of the code needs an inline function or a macro?

Reply:
(int32_t)buffer[0] & 0xff is repetitive. Also repeats in 64 bit. You can have a templated inline function to mask the value.

Reply:
But then the inline function call itself will be repetitive, wouldn't it apply to that also?

Reply:
Actually, let me try. I have never written a templated inline function before.